test/idToken.tests.js

```diff
@@ -0,0 +1,308 @@
+  auth_time: TODAY_IN_SECONDS
+};
+
+var defaultOptions = {
+  issuer: expectations.issuer,
+  audience: [expectations.clientId, expectations.clientIdAlt],
+  nonce: expectations.nonce
+};
+
+function generateJWT(bodyOverrides, alg) {
+  var body = Object.assign({}, payload, bodyOverrides || {});
+  alg = alg || 'RS256';
+  var options = {
+    header: { alg: alg },
+    payload: body
+  };
+  if (alg === 'RS256') {
+    options.privateKey = privateKeyRSA;
+  } else if (alg === 'HS256') {
+    options.secret = secretHMAC;
+  }
+  return jws.sign(options);
+}
+
+describe('idToken.decode', function() {
+  it('should decode a valid token', function() {
+    var alg = 'RS256';
+    var token = generateJWT({ name: 'ÁÁutf8' }, alg);
+    var decoded = idToken.decode(token);
+
+    assert.equal(decoded._raw, token);
+    assert.equal(decoded.header.alg, alg);
+
+    Object.keys(payload).forEach(function(key) {
+      assert.deepEqual(payload[key], decoded.payload[key]);
+    });
+  });
+
+  it('throws errors on invalid tokens', function() {
+    var IDTOKEN_ERROR_MESSAGE = 'ID token could not be decoded';
+
+    it('throws when there is more or less than 3 parts', function() {
+      assert.throws(idToken.decode('test'), IDTOKEN_ERROR_MESSAGE);
+      assert.throws(idToken.decode('test.'), IDTOKEN_ERROR_MESSAGE);
+      assert.throws(idToken.decode('test.test'), IDTOKEN_ERROR_MESSAGE);
+      assert.throws(idToken.decode('test.test.test.test'), IDTOKEN_ERROR_MESSAGE);
+    });
+    it('throws when there is no header', function() {
+      assert.throws(idToken.decode('.test.test'), IDTOKEN_ERROR_MESSAGE);
+    });
+    it('throws when there is no payload', function() {
+      assert.throws(idToken.decode('test..test'), IDTOKEN_ERROR_MESSAGE);
+    });
+    it('throws when there is no signature', function() {
+      assert.throws(idToken.decode('test.test.'), IDTOKEN_ERROR_MESSAGE);
+    });
+  });
+});
+
+describe('idToken.validate', function() {
+  var expectedOptions;
+
+  beforeEach(function() {
+    expectedOptions = Object.assign({}, defaultOptions);
+    expectedOptions.audience = expectations.clientId;
+    expectedOptions.maxAge = 123;
+  });
+
+  it('should throw when no id token is present', function(done) {
+    var EXPECTED_ERROR_MESSAGE = 'ID token is required but missing';
+    try {
+      idToken.validate();
+      done(new Error('Should have thrown error: ' + EXPECTED_ERROR_MESSAGE));
+    } catch (error) {
+      assert.equal(error.message, EXPECTED_ERROR_MESSAGE);
+      done();
+    }
+  });
+  it('should throw when no Issuer is present in the claim', function(done) {
+    var EXPECTED_ERROR_MESSAGE = 'Issuer (iss) claim must be a string present in the ID token';
+    try {
+      idToken.validate(generateJWT({ iss: undefined }));
+      done(new Error('Should have thrown error: ' + EXPECTED_ERROR_MESSAGE));
+    } catch (error) {
+      assert.equal(error.message, EXPECTED_ERROR_MESSAGE);
+      done();
+    }
+  });
```

Comment:
**Nit:** There's a distincy pattern emerging with these tests. It feels like much of this could be refactored into a helper function to reduce visual clutter.

Also, the inclusion of a `done` param in the function makes me think that the test is async, which isn't the case. I can see why it's being used, just wondering if it would be better written in such a way that we didn't need to call `done`.